rust/passwords/src/sha512crypt.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025 Arista Networks, Inc.
+// Use of this source code is governed by the Apache License 2.0
+// that can be found in the LICENSE file.
+
+use sha_crypt::{CryptError, Sha512Params, sha512_crypt_b64};
+
+pub fn sha512_crypt(password: String, salt: String) -> Result<String, CryptError> {
+    // Setting rounds to 5000 which is the default for sha512crypt
+    let params = Sha512Params::new(5_000).expect("Could not instantiate Sha512Params.");
+
+    // Hash the password
+    let hashed_password = sha512_crypt_b64(password.as_bytes(), salt.as_bytes(), &params)
+        .expect("Failed to compute sha512 password.");
```

Comment:
you are never returning a result here are you? You should use ? instead of expect. Same above. Expect will panic instead of returning an Err.

Reply:
https://docs.rs/sha-crypt/latest/sha_crypt/

thats the example usage